packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -256,140 +258,268 @@ export function isAssignable(
 ) {
-	const rootNode: CheckerTreeEntryNode<T> = {
-		possibleRuntimeValues: [],
+	const rootNode: CheckerTreeNode<T> = {
 		parent: undefined,
-		typeDef: typeDef,
+		runtimeKey: undefined,
+		possibleValues: [],
+		validDefinitions: [],
+		condensedErrors: [],
 	}
-	rootNode.possibleRuntimeValues = runtimeValues.map(n => ({
+	rootNode.possibleValues = runtimeValues.map(n => ({
 		node: n,
-		validDefinitions: [],
+		parent: undefined,
+		children: [],
+		entryNode: rootNode,
+	}))
+	const simplifiedRoot = simplify(typeDef, undefined, undefined, options)
+	const validRootDefinitions = simplifiedRoot.kind === 'union'
+		? simplifiedRoot.members
+		: [simplifiedRoot]
+	rootNode.validDefinitions = validRootDefinitions.map(d => ({
+		parent: undefined,
 		entryNode: rootNode,
+		typeDef: d,
 	}))
-	const nodeQueue: CheckerTreeEntryNode<T>[] = [rootNode]
+	const nodeQueue: CheckerTreeNode<T>[] = [rootNode]
 
 	while (nodeQueue.length !== 0) {
 		const node = nodeQueue.splice(0, 1)[0]
-		check(node, options)
-		for (const runtimeValue of node.possibleRuntimeValues) {
-			let siblingErrors = runtimeValue.validDefinitions.map(d => ({
-				node: d,
-				errors: d.errors,
-			}))
-			let parent: CheckerTreeRuntimeNode<T> | undefined = runtimeValue
-			let depth = 0
-			while (parent) {
-				const { siblings, condensedErrors } =
-					condenseErrorsAndFilterSiblings(siblingErrors)
-				// TODO possible optimization: Remove entries from nodeQueue which are now no longer neccessary to evaluate.
-				// This is quite tricky and would mess up the check at the bottom here, and maybe not worth it if this is no bottleneck
-				parent.validDefinitions = siblings
-				parent.condensedChildErrors ??= []
-				parent.condensedChildErrors.push(condensedErrors)
-
-				// TypeScript is drunken (no reason for typedef here)
-				const oldParent: CheckerTreeRuntimeNode<T> = parent
-				parent = oldParent.entryNode.parent?.runtimeNode
-
-				const lastDefinition = parent
-					?.validDefinitions[parent.validDefinitions.length - 1]
-				const lastChild = lastDefinition?.children.findLast(c => {
-					let values = c.possibleRuntimeValues
-					for (let i = 0; i < depth; i++) {
-						values = values.flatMap(v => v.validDefinitions).flatMap(v =>
-							v.children
-						).flatMap(v => v.possibleRuntimeValues)
-					}
-					return values.length > 0
-				})
-				const lastValue = lastChild?.possibleRuntimeValues.findLast(v => {
-					let values = [v]
-					for (let i = 0; i < depth; i++) {
-						values = values.flatMap(v => v.validDefinitions).flatMap(v =>
-							v.children
-						).flatMap(v => v.possibleRuntimeValues)
-					}
-					return values.length > 0
-				})
+		const encounterdErrors: CheckerTreeError<T>[] = []
 
-				if (lastValue !== oldParent) {
-					// Wait for all siblings to be evaluated first
-					break
+		for (const value of node.possibleValues) {
+			const inferredSimplified = simplify(
+				value.node.inferredType,
+				value.entryNode.runtimeKey,
```

Comment:
This can be replaced with `node.runtimeKey` I believe

Reply:
Yes, it can. Vice versa the line below could be `value.entryNode.parent`, but I agree it makes more sense to change this one.